### PR TITLE
CORE-13126. Create Test_NtFreeVirtualMemory_Parameters().

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Add an ApiTest, before fixing the actual code.

JIRA issue: [CORE-13126](https://jira.reactos.org/browse/CORE-13126)

## Proposed changes

- Code formatting fixes.
- Create `Test_NtFreeVirtualMemory_Parameters()`.

## TODO

- [X] Needs to be tested on WS03. (I can't do it. It passes on WXP.) // Done by @binarymaster.

### CODE REVIEW

**Comment by SergeGautherie** [comment]
@ThFabba, already involved in Jira issue.

**Comment by HBelusca** [comment]
In well-written English one would say: "Code style fixes" or "Code formatting fixes" instead of "style nits".

File: modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c [hunk 1]
```diff
@@ -10,7 +10,7 @@ static void Test_NtFreeVirtualMemory(void)
     PVOID Buffer = NULL, Buffer2;
     SIZE_T Length = PAGE_SIZE;
     NTSTATUS Status;
-    
+
```
Comment (learn-more) [inline modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c]: Why?

File: modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c [hunk 2]
```diff
@@ -167,8 +167,48 @@ static void Test_NtFreeVirtualMemory(void)
...
 }
-    
+
+static void Test_NtFreeVirtualMemory_Parameters(void)
+{
+    NTSTATUS Status;
+    ULONG FreeType;
+    int i;
+
+    // 4th parameter: "ULONG FreeType".
```
Comment (learn-more) [inline modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c]: Leftover?

File: modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c [hunk 3]
```diff
@@ -167,8 +167,48 @@ static void Test_NtFreeVirtualMemory(void)
...
+    {
+        FreeType = 1 << i;
+        if (FreeType == MEM_DECOMMIT || FreeType == MEM_RELEASE)
+            continue;
+
+        Status = NtFreeVirtualMemory(NULL, NULL, NULL, FreeType);
+        ok(Status == STATUS_INVALID_PARAMETER_4, "NtFreeVirtualMemory returned status : 0x%08lx\n", Status);
+    }
+    // All bits at once.
+    // Not (fuzzy-)testing all other values.
```
Comment (learn-more) [inline modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c]: Writing (sentences) like (this) is not making them (more) readable.

File: modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c (hunk 1 above)
Comment (SergeGautherie) [reply modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c]: Indentation of blank lines seems "unusual".
I split that part out in a second commit: to be squashed or discarded?

File: modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c (hunk 2 above)
Comment (SergeGautherie) [reply modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c]: No, I meant it as a "header". Wrt possibly testing other parameters too.

File: modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c (hunk 3 above)
Comment (SergeGautherie) [reply modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c]: Removed "(fuzzy-)".

**Comment by SergeGautherie** [comment]
> "Code formatting fixes"

Thanks. Used that.

**Pull request merged**